### PR TITLE
Add activeThermostatMode to thermostat query response

### PR DESCRIPTION
**Pull request by rukuh**
## :recycle: Current situation

_Describe the current situation. Explain current problems, if there are any. Be as descriptive as possible (e.g., including examples or code snippets)._

In the Google Home app, when a Nest Thermostat is either heating or cooling the device box is highlighted orange or blue to indicate actively heating or cooling. Custom thermostat plugins do not show colors based on whether the device is actively heating/cooling.

## :bulb: Proposed solution

_Describe the proposed solution and changes. How does it affect the project? How does it affect the internal structure (e.g., refactorings)?_

Add `activeThermostatMode` based on Google Home developer docs (https://developers.home.google.com/cloud-to-cloud/traits/temperaturesetting)

## :gear: Release Notes

_Provide a summary of the changes or features from a user's point of view. If there are breaking changes, provide migration guides using code examples of the affected features._

Custom thermostat plugins will now show active heating or cooling state in the Google Home app.

## :heavy_plus_sign: Additional Information

_If applicable, provide additional context in this section._

### Testing

_Which tests were added? Which existing tests were adapted/changed? Which situations are covered, and what edge cases are missing?_

Unit tests for thermostat type were updated to include `activeThermostatMode` in the expected response.

### Reviewer Nudging

_Where should the reviewer start? what is a good entry point?_

Run a custom plugin for a thermostat and just statically set the state in the Google Home app.

### CODE REVIEW

**Comment by NorthernMan54** [comment]
Tks for this, I'm in the middle of making other changes, and can merge this into a beta in a few days

**Review by NorthernMan54 — DISMISSED** [review]

File: src/types/thermostat.ts [hunk 1]
```diff
@@ -38,11 +38,14 @@ export class Thermostat extends ghToHap implements ghToHap_t {
   }
 
   query(service: ServiceType) {
+    const currentHeatingCoolingState: number = Number(service.serviceCharacteristics.find(x => x.uuid === Characteristic.CurrentHeatingCoolingState).value);
+    const activeThermostatMode = ['off', 'heat', 'cool'][currentHeatingCoolingState];
```
[the remark this answers is not in the document]
Comment (NorthernMan54) [reply src/types/thermostat.ts:42]: @rukuh  Are you okay with this ?

**Review by NorthernMan54 — APPROVED** [review]

**Comment by NorthernMan54** [comment]
Published in beta-4.0.1
